apiconcepts/integration/adding_custom_wizard_steps.md

```diff
@@ -0,0 +1,79 @@
+
+### Data
+* You can define custom data for your pages using the `Data` property on the [StudioWizardPage](../../api/integration/Sdl.Desktop.IntegrationApi.Wizard.StudioWizardPage.yml) object.
+* The Data property is a `Dictionary<string,object>` that is common to all wizard pages. 
+* Once a dictionary key is set from a page, it becomes available to any pages that are displayed after it. This includes pages that are navigated to via the **Back** button. 
+* You can use the Page's `OnShow` method to query for the most recent key values that may have been updated/inserted by previously shown pages.
+* The `Data` property is initialized for all pages before the wizard is displayed.
+
+### Icon
+* Set this property to display an icon in the top-right area of the page.
+* Set this property to `null` if you don't want an icon displayed.
+
+### Id
+* This property is used by the wizard framework to uniquely identify the pages.
+
+  
+# Integrating pages as steps into <Var:ProductName> wizards
+Custom pages are injected into the wizard via the `firstPages` argument of the event object corresponding to the respective wizard. Keep in mind that the `firstPages` argument is available only for some wizard event objects.
+ 
+To inject the custom pages as steps into a <Var:ProductName> wizard:
+
+* Determine or define a place in your code to launch the wizard (with custom steps) from; for example, from an [Action](../../api/integration/Sdl.Desktop.IntegrationApi.AbstractAction.yml)'s `Execute` method.
+* Determine the appropriate event (e.g. [OpenProjectPackageEvent](../../api/integration/Sdl.TranslationStudioAutomation.IntegrationApi.Events.OpenProjectPackageEvent.yml)) to be raised, that corresponds to the wizard you want to open.
+* Set up a `List` of [StudioWizardPage](../../api/integration/Sdl.Desktop.IntegrationApi.Wizard.StudioWizardPage.yml) instances indicating the custom pages to be inserted
+* Create the event, passing in `firstPages` list as an argument.
+* Use the event aggregator to raise the event. This will launch the wizard.
+
+Notes:
+* The custom pages appear in the wizard at the beginning, before any other standard pages.
+* The custom pages appear in the wizard in the same order they are defined in the list configured in the event object.
+
+Wizard conceptual execution flow
+----
+The diagram below briefly describes the wizard's execution sequence, and shows the order in which `StudioWizardPage` properties and methods are called by the framework at runtime.
+
+<img style="display:block; " src="images/Wizard public API flow.png"/>
+
+Open package wizard specifics
+----
+* You can provide your own custom package converter to be used in the Open Package wizard. To do so, create a class that implements the `IExternalPackageConverter` interface and decorate it with the `ExternalPackageConvertor` attribute. The Plugin Framework will pick it up as an extension point.
```

Comment:
Nasty passive voice. I would use: You can use your own custom package converter in the Open Package wizard by creating a class that ....

Reply:
I ended up rewriting the whole paragraph, to improve clarity.